app/services/foreman/renderer/scope/macros/snippet_rendering.rb

```diff
@@ -46,9 +62,15 @@ def snippet_if_exists(name, options = {})
             example "snippet('ntp_server', variables: { :enable_ntp => true) } # => '...'"
           end
           def snippet(name, options = {}, variables: {})
-            template = source.find_snippet(name)
+            if name.is_a?(Array)
+              tpl_name = name.find { |tpl| snippet_exists(tpl) } || ''
+            else
+              tpl_name = name
+            end
+
+            template = source.find_snippet(tpl_name)
             unless template
-              raise "The specified snippet '#{name}' does not exist, or is not a snippet." unless options[:silent]
+              raise "The specified snippet '#{tpl_name}' does not exist, or is not a snippet." unless options[:silent]
```

Comment:
I don't think this is right. If `name` is an array, this will be an empty string.

Reply:
Good catch, I missed this one.